### PR TITLE
Providers

### PR DESCRIPTION
**Pull request by bennypowers**
This PR adds a `providers` property, which users can use to define idP's in AWS credentials Logins map.

It also refactors the code which deals with credentials

### CODE REVIEW

**Comment by bennypowers** [comment]
This isn't quite ready yet. I think adding a provider should toggle loggedIn

**Comment by bennypowers** [comment]
Ok this is ready to fly at your discretion

File: aws-cognito.html [hunk 1]
```diff
@@ -72,11 +82,29 @@
         },
 
         /**
-         * The currently authenticated user, with attributes.
+         * cognitoUser object.
          */
         user: {
           type: Object,
-          readOnly: true,
```
Comment (gjdenhertog) [inline aws-cognito.html:79]: Shouldn't this still be readOnly for the outside world?

Comment (bennypowers) [reply aws-cognito.html:79]: Computed properties are read only see https://github.com/Polymer/polymer/issues/1925

**Pull request merged**